src/Distribution/Audit.hs

```diff
@@ -194,6 +208,80 @@ osvHandler mkHandle mp =
               ]
         ]
 
+sarifHandler
+  :: MonadUnliftIO m
+  => Codensity IO Handle
+  -> ProjectBaseContext
+  -> [(PackageName, ElaboratedPackageInfoAdvised)]
+  -> m ()
+sarifHandler mkHandle projectBaseContext packageAdvisories = do
+  let projectRoot = distProjectRootDirectory $ distDirLayout projectBaseContext -- TODO(blackheaven): resolve repository root
```

Comment:
ah yeah, I see you have left a `TODO` here, yes, it would be good if we would hand this information not as a whole but only what we need :)